src/raphson.jl

```diff
@@ -131,9 +143,23 @@ function SciMLBase.__init(prob::NonlinearProblem{uType, iip}, alg::NewtonRaphson
                           args...;
                           alias_u0 = false,
                           maxiters = 1000,
-                          abstol = 1e-6,
+                          abstol = nothing,
+                          reltol = nothing,
                           internalnorm = DEFAULT_NORM,
                           kwargs...) where {uType, iip}
+    tc = alg.termination_condition
+    ueltype = eltype(uType)
+    abstol = !isnothing(abstol) ? abstol :
+             (!isnothing(tc.abstol) ? tc.abstol :
+              real(oneunit(ueltype)) * (eps(real(one(ueltype))))^(4 // 5))
+
+    reltol = !isnothing(reltol) ? reltol :
+             (!isnothing(tc.reltol) ? tc.reltol : eps(real(one(ueltype)))^(4 // 5))
+
+    mode = DiffEqBase.get_termination_mode(tc)
+    storage = mode ∈ DiffEqBase.SAFE_TERMINATION_MODES ? Dict() : nothing
```

Comment:
Not sure. I took it from https://github.com/SciML/SimpleNonlinearSolve.jl/blob/df85c3f9b98bd48aeb07ad15754303f7bffb3a3d/src/broyden.jl#L51

Reply:
@avik-pal

Reply:
It is needed since the closure can only return boolean (so that it works directly with Terminating Callbacks), and we might need additional information about the termination state (https://github.com/SciML/DiffEqBase.jl/blob/77c713880b5ab15b084824f0649e20d15db528e7/src/termination_conditions.jl#L168) or other values (for the `Best` modes)

Reply:
Yeah we could just create a struct for that purpose. I had just copied the implementation over from one of my research codebases so it might not have been too polished

Reply:
We should probably do that before adding this. The dictionary isn't something we'll want to keep.

Reply:
So who will be doing that refactoring?  Is it @avik-pal ?

Reply:
I can take a look into it early next week

Reply:
https://github.com/SciML/DiffEqBase.jl/pull/903